ledger-api/grpc-definitions/com/daml/ledger/api/v2/update_service.proto

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2023 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+syntax = "proto3";
+
+package com.daml.ledger.api.v2;
+
+import "com/daml/ledger/api/v1/ledger_offset.proto";
+import "com/daml/ledger/api/v2/transaction.proto";
+import "com/daml/ledger/api/v2/transaction_filter.proto";
+import "com/daml/ledger/api/v2/reassignment.proto";
+
+option java_outer_classname = "UpdateServiceOuterClass";
+option java_package = "com.daml.ledger.api.v2";
+option csharp_namespace = "Com.Daml.Ledger.Api.V2";
+
+// TODO add documentation of participant offsets and virtual shared ledger and causality guarantees
+// Allows clients to read updates (transactions and reassignments) from the ledger.
+service UpdateService {
+
+  // Read the ledger's filtered transaction stream and related reassignments for a set of parties.
+  // For transactions it lists only creates and archives, but not other events.
+  // Omits all events on transient contracts, i.e., contracts that were both created and archived in the same transaction.
+  rpc GetUpdates (GetUpdatesRequest) returns (stream GetUpdatesResponse);
+
+  // Read the ledger's complete transaction tree stream and related reassignments for a set of parties.
+  // The stream will be filtered only by the parties as wildcard parties.
+  // The template/interface filters describe the respective fields in the ``CreatedEvent`` results.
+  rpc GetUpdateTrees (GetUpdatesRequest) returns (stream GetUpdateTreesResponse);
+
+  // Lookup a transaction tree by the ID of an event that appears within it.
+  // For looking up a transaction instead of a transaction tree, please see GetFlatTransactionByEventId
+  rpc GetTransactionByEventId (GetTransactionByEventIdRequest) returns (GetTransactionResponse);
+
+  // Lookup a transaction tree by its ID.
+  // For looking up a transaction instead of a transaction tree, please see GetFlatTransactionById
+  rpc GetTransactionById (GetTransactionByIdRequest) returns (GetTransactionResponse);
+
+  // Lookup a transaction by the ID of an event that appears within it.
+  rpc GetFlatTransactionByEventId (GetTransactionByEventIdRequest) returns (GetFlatTransactionResponse);
+
+  // Lookup a transaction by its ID.
+  rpc GetFlatTransactionById (GetTransactionByIdRequest) returns (GetFlatTransactionResponse);
```

Comment:
can you remind me what we discussed regarding ReassignmentById? Might add later but not in the first version?

Reply:
AFAIR: We discussed that we wont complete the update and command services wrt reassignments, because we do not want to commit to implement these prematurely for 3.0. (also not needed for testnet, not planned to be implemented, would be confusing to add it not and not implement it)...I also retracted my changes for making it complete accordingly.

Reply:
wfm